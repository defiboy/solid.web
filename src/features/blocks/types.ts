import { Transaction } from "features/transactions";

export interface BlockHeader {
    number: number
    hash: string
    parentHash: string
    nonce: string
    sha3Uncles: string
    logsBloom: string
    transactionRoot: string
    stateRoot: string
    receiptRoot: string
    miner: string
    extraData: string
    gasLimit: number
    gasUsed: number
    timestamp: number | string
}

export interface Block extends BlockHeader {
    transactions: Transaction[];
    size: number
    difficulty: number
    totalDifficulty: number
    uncles: string[];
}