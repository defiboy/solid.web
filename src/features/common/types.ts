export enum Status { // TODO, with sagas I think it won't apply
    InProgress = 'InProgress',
    Completed = 'Completed',
    Synchronizing = 'Synchronizing', // this won't apply for connection modal
    Failed = 'Failed',
    NotStarted = 'NotStarted',
    Started = 'Started'
}

export interface GenericResponse<T> {
    total: number;
    limit: number;
    skip: number;
    data: T;
}

export interface GenericArrayResponse<T> {
    total: number;
    limit: number;
    skip: number;
    data: T[];
}
