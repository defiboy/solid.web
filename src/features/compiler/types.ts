import { Status } from '../common/types'

export interface ValidateSourceCode {
  status: Status
  compilerVersion: string
  sourceCode: string
}

export interface LoadCompilerRequest {
  version: string
  status: Status
}
