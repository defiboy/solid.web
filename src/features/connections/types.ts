import { Connection } from '@solid-explorer/types'

export interface ConnectionItem extends Connection {
  type: string
}
