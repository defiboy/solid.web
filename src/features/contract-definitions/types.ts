import { ContractDefinition } from '@solidstudio/solid.types'

export interface ContractDefinitionItem extends ContractDefinition {
    type: string
}