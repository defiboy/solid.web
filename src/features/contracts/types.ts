import { Contract } from '@solid-explorer/types'

export interface ContractItem extends Contract {
  type: 'editor' // TODO IMPROVE
}

export interface ContractNormalized extends Contract {
  traces?: string[]
}

export interface Storage {
  slot: string
  value: string
  name?: string
}