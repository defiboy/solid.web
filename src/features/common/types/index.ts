export enum Status { // TODO, with sagas I think it won't apply
    InProgress = 'InProgress',
    Completed = 'Completed',
    Synchronizing = 'Synchronizing', // this won't apply for connection modal
    Failed = 'Failed',
    NotStarted = 'NotStarted',
    Started = 'Started'
}