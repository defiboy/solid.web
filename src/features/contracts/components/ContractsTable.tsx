import React from 'react'

import { ColumnProps } from 'antd/es/table';

import { Contract } from '../types';
import { ContractTableComponent } from './ContractsTableComponent';

interface OwnProps {
    contracts?: Contract[]
    connectionId: string | undefined
}

type AllProps = OwnProps //& DispatchProps & StateProps

const tableColumns: ColumnProps<Contract>[] = [
    {
        key: 'name',
        title: 'Contract Name',
        dataIndex: 'name',
    },
    {
        key: 'address',
        title: 'Contract Address',
        dataIndex: 'address',
    },
    {
        key: 'transactionCount',
        title: 'Transaction Count',
        dataIndex: 'transactionCount',
    },
    {
        key: 'creationDate',
        title: 'Creation Date',
        dataIndex: 'creationDate',
    },
    {
        key: 'lastExecutionDate',
        title: 'Last Execution Date',
        dataIndex: 'lastExecutionDate',
    }

];

export class ContractsTable extends React.Component<AllProps> {
    render() {
        const { contracts } = this.props
        return <ContractTableComponent dataSource={contracts} columns={tableColumns} />
    }
}