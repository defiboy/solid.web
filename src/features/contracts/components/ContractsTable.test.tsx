import React from 'react';
import { render, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'; // TO REMOVE

import { Contract, buildFakeContracts } from '@solidstudio/solid.types'
import { ContractsTable } from './ContractsTable'

describe('Contracts table test', () => {

    let contracts: Contract[]
    const mockOnClickTable = jest.fn()
    const mockOnDoubleClickTable = jest.fn()

    const renderBlocksTable = (contracts: Contract[] | undefined) => {
        return render(
            <ContractsTable contracts={contracts} onClick={mockOnClickTable} onDoubleClick={mockOnDoubleClickTable} />
        )
    }

    beforeEach(() => {
        contracts = buildFakeContracts()
        mockOnDoubleClickTable.mockReset()
        mockOnClickTable.mockReset()
    })

    test('that table renders correctly with data', () => {
        const { getByTestId } = renderBlocksTable(contracts)

        contracts.forEach((item) => {
            const row = getByTestId(`contracts-table-row-${item.address}`)
            expect(row).toBeInTheDocument()
        })
    })

    test('that table renders correctly without data', () => {
        const { getByText } = renderBlocksTable(undefined)

        const noDataInTable = getByText(/no data/i)

        expect(noDataInTable).toBeInTheDocument()
    })

    test('snapshot', () => {
        const { container } = renderBlocksTable(contracts)

        expect(container).toMatchSnapshot()
    })

    test('clicks', () => {
        const { getByTestId } = renderBlocksTable(contracts)

        contracts.forEach((item) => {
            const row = getByTestId(`contracts-table-row-${item.address}`)

            fireEvent.click(row)

            expect(mockOnClickTable).toHaveBeenCalledTimes(1)
            expect(mockOnClickTable).toHaveBeenCalledWith(item)
            expect(mockOnDoubleClickTable).toHaveBeenCalledTimes(0)
            mockOnClickTable.mockClear()
        })
    })

    test('double clicks', () => {
        const { getByTestId } = renderBlocksTable(contracts)

        contracts.forEach((item) => {
            const row = getByTestId(`contracts-table-row-${item.address}`)

            fireEvent.doubleClick(row)

            expect(mockOnDoubleClickTable).toHaveBeenCalledTimes(1)
            expect(mockOnDoubleClickTable).toHaveBeenCalledWith(item)
            expect(mockOnClickTable).toHaveBeenCalledTimes(0)
            mockOnDoubleClickTable.mockClear()
        })
    })
})