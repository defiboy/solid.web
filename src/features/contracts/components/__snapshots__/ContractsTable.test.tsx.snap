// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contracts table test snapshot 1`] = `
<div>
  <div
    class="ant-table-wrapper"
  >
    <div
      class="ant-spin-nested-loading"
    >
      <div
        class="ant-spin-container"
      >
        <div
          class="ant-table ant-table-default ant-table-scroll-position-left"
        >
          <div
            class="ant-table-content"
          >
            <div
              class="ant-table-body"
            >
              <table
                class=""
              >
                <colgroup>
                  <col />
                  <col />
                  <col />
                  <col />
                  <col />
                </colgroup>
                <thead
                  class="ant-table-thead"
                >
                  <tr>
                    <th
                      class=""
                    >
                      <span
                        class="ant-table-header-column"
                      >
                        <div>
                          <span
                            class="ant-table-column-title"
                          >
                            Contract Name
                          </span>
                          <span
                            class="ant-table-column-sorter"
                          />
                        </div>
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span
                        class="ant-table-header-column"
                      >
                        <div>
                          <span
                            class="ant-table-column-title"
                          >
                            Contract Address
                          </span>
                          <span
                            class="ant-table-column-sorter"
                          />
                        </div>
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span
                        class="ant-table-header-column"
                      >
                        <div>
                          <span
                            class="ant-table-column-title"
                          >
                            Transaction Count
                          </span>
                          <span
                            class="ant-table-column-sorter"
                          />
                        </div>
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span
                        class="ant-table-header-column"
                      >
                        <div>
                          <span
                            class="ant-table-column-title"
                          >
                            Creation Date
                          </span>
                          <span
                            class="ant-table-column-sorter"
                          />
                        </div>
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span
                        class="ant-table-header-column"
                      >
                        <div>
                          <span
                            class="ant-table-column-title"
                          >
                            Last Execution Date
                          </span>
                          <span
                            class="ant-table-column-sorter"
                          />
                        </div>
                      </span>
                    </th>
                  </tr>
                </thead>
                <tbody
                  class="ant-table-tbody"
                >
                  <tr
                    class="ant-table-row ant-table-row-level-0"
                    data-row-key="0xAC716460A84B85d774bEa75666ddf0088b024741"
                  >
                    <td
                      class=""
                    >
                      ERC-20
                    </td>
                    <td
                      class=""
                    >
                      <p
                        data-testid="contracts-table-row-0xAC716460A84B85d774bEa75666ddf0088b024741"
                      >
                        0xAC716460A84B85d774bEa75666ddf0088b024741
                      </p>
                    </td>
                    <td
                      class=""
                    >
                      1
                    </td>
                    <td
                      class=""
                    >
                      1/19/2020
                    </td>
                    <td
                      class=""
                    >
                      1/19/2020
                    </td>
                  </tr>
                  <tr
                    class="ant-table-row ant-table-row-level-0"
                    data-row-key="0xAC716460A84B85d774bEa75666ddf0088b024742"
                  >
                    <td
                      class=""
                    >
                      ERC-721
                    </td>
                    <td
                      class=""
                    >
                      <p
                        data-testid="contracts-table-row-0xAC716460A84B85d774bEa75666ddf0088b024742"
                      >
                        0xAC716460A84B85d774bEa75666ddf0088b024742
                      </p>
                    </td>
                    <td
                      class=""
                    >
                      1
                    </td>
                    <td
                      class=""
                    >
                      1/19/2020
                    </td>
                    <td
                      class=""
                    >
                      1/19/2020
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
        <ul
          class="ant-pagination ant-table-pagination"
          unselectable="unselectable"
        >
          <li
            aria-disabled="true"
            class="ant-pagination-disabled ant-pagination-prev"
            title="Previous Page"
          >
            <a
              class="ant-pagination-item-link"
            >
              <i
                aria-label="icon: left"
                class="anticon anticon-left"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  data-icon="left"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 0 0 0 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                  />
                </svg>
              </i>
            </a>
          </li>
          <li
            class="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
            tabindex="0"
            title="1"
          >
            <a>
              1
            </a>
          </li>
          <li
            aria-disabled="true"
            class="ant-pagination-disabled ant-pagination-next"
            title="Next Page"
          >
            <a
              class="ant-pagination-item-link"
            >
              <i
                aria-label="icon: right"
                class="anticon anticon-right"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  data-icon="right"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M765.7 486.8L314.9 134.7A7.97 7.97 0 0 0 302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 0 0 0-50.4z"
                  />
                </svg>
              </i>
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
